File: A1/A1-023/A1-023.py
# --------------------------------------------------
# File Name : A1-023.py
# Problem   : Water State
# Author    : Worralop Srichainont
# Date      : 2026-02-08
# --------------------------------------------------

# Input temperature
temperature = int(input())
unit = input().strip()

# Output the water state for Celsius unit
if unit.upper() == "C":
    if temperature <= 0:
        print("solid")
    elif temperature < 100:
        print("liquid")
    elif temperature >= 100:
        print("gas")

# Output the water state for Fahrenheit unit
if unit.upper() == "F":
    if temperature <= 32:
        print("solid")
    elif temperature < 212:
        print("liquid")
    elif temperature > 212:
        print("gas")

File: A1/A1-023/A1-023.cpp
#include<iostream>
using namespace std;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);
    
    // Input
    int temp;
    char unit;
    cin >> temp >> unit;
    // Output
    if(unit == 'C' || unit == 'c')
    {
        if(temp <= 0) { cout << "solid"; }
        else if(temp < 100) { cout << "liquid"; }
        else { cout << "gas"; }
    }
    else if(unit == 'F' || unit == 'f')
    {
        if(temp <= 32) { cout << "solid"; }
        else if(temp < 212) { cout << "liquid"; }
        else { cout << "gas"; }
    }
    return 0;
}